packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -689,19 +689,26 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
      * @param validator Address of the validator.
      */
     function bondValidatorKro(address validator) external onlyValidatorManager {
-        Vault storage vault = _vaults[validator];
-        if (vault.asset.validatorKro - vault.asset.validatorKroBonded < BOND_AMOUNT)
-            revert InsufficientAsset();
+        Asset storage asset = _vaults[validator].asset;
+        uint128 remainder = asset.validatorKro - asset.validatorKroBonded;
+        if (remainder < BOND_AMOUNT) revert InsufficientAsset();
 
         unchecked {
-            vault.asset.validatorKroBonded += BOND_AMOUNT;
+            asset.validatorKroBonded += BOND_AMOUNT;
         }
 
-        emit ValidatorKroBonded(
-            validator,
-            BOND_AMOUNT,
-            vault.asset.validatorKro - vault.asset.validatorKroBonded
-        );
+        emit ValidatorKroBonded(validator, BOND_AMOUNT, remainder - BOND_AMOUNT);
+    }
+
+    /**
+     * @notice Unbond KRO from validator KRO during output finalization or challenge slashing. This
+     *         function is only called by the ValidatorManager contract.
+     *
+     * @param validator Address of the validator.
+     */
+    function unbondValidatorKro(address validator) external onlyValidatorManager {
+        Asset storage asset = _vaults[validator].asset;
+        _unbondValidatorKro(validator, asset);
```

Comment:
Let's emit event outside of `_unbondValidatorKro`. I think it would be better to have the state change happen inside the internal function, and the event emission for the associated the state change happen in the original function.

Reply:
Then I think it's better to just remove the internal function. I'll apply it